src/main/java/plugily/projects/buildbattle/handlers/menu/registry/particles/ParticleRefreshScheduler.java

```diff
@@ -40,29 +40,38 @@ public class ParticleRefreshScheduler {
   private BukkitTask task;
 
   public ParticleRefreshScheduler(PluginMain plugin) {
-    if(task != null) {
-      task.cancel();
+    int particleRefreshTick = plugin.getConfig().getInt("Particle-Refresh-Per-Tick", 10);
+
+    if (particleRefreshTick < 0) {
+      particleRefreshTick = 20;
     }
 
-    int particleRefreshTick = plugin.getConfig().getInt("Particle-Refresh-Per-Tick", 10);
     int amountParticle = plugin.getConfig().getInt("Amount-One-Particle-Effect-Contains", 20);
 
+    if (amountParticle < 0) {
+      amountParticle = 20;
+    }
+
+    final int ap = amountParticle;
```

Comment:
ap -> particleAmount 
is final required? (it wasn't before? at the end yea its final from that on)

Reply:
It should be `final` as the `amountParticle` was changed above this and this variable defined in an enclosing scope.